src/2013.ipynb

```diff
@@ -150,26 +142,26 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 33,
+   "execution_count": 10,
    "metadata": {},
    "outputs": [],
    "source": [
     "regex_schema = {\n",
     "    'title': r'Project Title:(.*?)Name',\n",
     "    'students': r'Name:(.*?)Programme',\n",
     "    'programme': r'Programme:(.*)Supervisor',\n",
-    "    'supervisor_and_description': r'Supervisor:(.*)Primary Area',\n",
-    "    'primary_area': r'Primary Area:(.*)Secondary Area',\n",
-    "    'secondary_area': r'Secondary Area:(.*)Primary OS',\n",
-    "    'primary_os': r'Primary OS:(.*?)Primary Technology',\n",
+    "    'supervisor_and_description': r'Supervisor:(.*?)(?:Primary Area:|Secondary Area:|Primary OS:|Primary Technology:|Secondary Technology:|$)',\n",
```

Comment:
These are a bit long regexes. But basically it's pretty random whether a primary area / secondary area / primary OS / primary tech / secondary tech is included in the project or not.
We just try to stop at the first field we find, and if we don't find it we try to stop at the next field.